akka-http/src/main/scala/akka/http/javadsl/server/directives/BasicDirectives.scala

```diff
@@ -53,6 +64,54 @@ abstract class BasicDirectives {
     D.mapResponseHeaders(l ⇒ Util.immutableSeq(f.apply(Util.javaArrayList(l))).map(_.asScala)) { inner.get.delegate } // TODO try to remove map()
   }
 
+  def mapInnerRoute(f: JFunction[Route, Route], inner: Supplier[Route]): Route = RouteAdapter {
+    D.mapInnerRoute(route => f(RouteAdapter(route)).delegate) { inner.get.delegate }
+  }
+
+  def mapRouteResult(f: JFunction[RouteResult, RouteResult], inner: Supplier[Route]): Route = RouteAdapter {
+    D.mapRouteResult(route => f(route)) { inner.get.delegate }
+  }
+
+  @CorrespondsTo("mapRouteResultFuture")
+  def mapRouteResultStage(f: JFunction[CompletionStage[RouteResult], CompletionStage[RouteResult]], inner: Supplier[Route]): Route = RouteAdapter {
```

Comment:
Isn't that weird when it does nothing with `Future`, it's an implementation detail that it does on the inside?